instrumentation-genai/opentelemetry-instrumentation-google-genai/src/opentelemetry/instrumentation/google_genai/tool_call_wrapper.py

```diff
@@ -0,0 +1,190 @@
+    return f"tool_call {function_name}"
+
+
+def _create_function_span_attributes(
+    wrapped_function, function_args, function_kwargs, extra_span_attributes
+):
+    """Creates the attributes for a tool call function span."""
+    result = {}
+    if extra_span_attributes:
+        result.update(extra_span_attributes)
+    result[code_attributes.CODE_FUNCTION_NAME] = wrapped_function.__name__
+    result["code.module"] = wrapped_function.__module__
+    result["code.args.positional.count"] = len(function_args)
+    result["code.args.keyword.count"] = len(function_kwargs)
+    return result
+
+
+def _record_function_call_argument(
+    span, param_name, param_value, include_values
+):
+    attribute_prefix = f"code.function.parameters.{param_name}"
+    type_attribute = f"{attribute_prefix}.type"
+    span.set_attribute(type_attribute, type(param_value).__name__)
+    if include_values:
+        value_attribute = f"{attribute_prefix}.value"
+        span.set_attribute(value_attribute, _to_otel_value(param_value))
+
+
+def _record_function_call_arguments(
+    otel_wrapper, wrapped_function, function_args, function_kwargs
+):
+    """Records the details about a function invocation as span attributes."""
+    include_values = is_content_recording_enabled()
+    span = trace.get_current_span()
+    signature = inspect.signature(wrapped_function)
+    params = list(signature.parameters.values())
+    for index, entry in enumerate(function_args):
+        param_name = f"args[{index}]"
+        if index < len(params):
+            param_name = params[index].name
+        _record_function_call_argument(span, param_name, entry, include_values)
+    for key, value in function_kwargs.items():
+        _record_function_call_argument(span, key, value, include_values)
+
+
+def _record_function_call_result(otel_wrapper, wrapped_function, result):
+    """Records the details about a function result as span attributes."""
+    include_values = is_content_recording_enabled()
+    span = trace.get_current_span()
+    span.set_attribute("code.function.return.type", type(result).__name__)
+    if include_values:
+        span.set_attribute(
+            "code.function.return.value", _to_otel_value(result)
+        )
+
+
+def _wrap_sync_tool_function(
+    tool_function: ToolFunction,
+    otel_wrapper: OTelWrapper,
+    extra_span_attributes: Optional[dict[str, str]] = None,
+    **kwargs,
+):
+    @functools.wraps(tool_function)
+    def wrapped_function(*args, **kwargs):
+        span_name = _create_function_span_name(tool_function)
+        attributes = _create_function_span_attributes(
+            tool_function, args, kwargs, extra_span_attributes
+        )
+        with otel_wrapper.start_as_current_span(
+            span_name, attributes=attributes
+        ):
+            _record_function_call_arguments(
+                otel_wrapper, tool_function, args, kwargs
+            )
+            result = tool_function(*args, **kwargs)
```

Comment:
See below comment but I think we are just ignoring the outer provided kwargs since it gets shadowed on L119

Reply:
Same as above.